File: funciones.js
// Multiplicación de Elementos de Dos Arreglos
let a = [1, 2, 3, 4, 5];
let b = [6, 7, 8, 9, 10];
let c = [];

for (let i = 0; i < a.length; i++) {
    c.push(a[i] * b[i]);
}
console.log(c);

// Funciones

// Definición y llamada de función para mostrar un texto
function mostrarTexto() {
    console.log('hola');
}
mostrarTexto();

// Función que multiplica dos valores dentro de la función
function multiplicar() {
    let a = 5;
    let b = 8;
    console.log(a * b);
}
multiplicar();

// Acceso a variables globales dentro de una función
let aGlobal = 5;
let bGlobal = 8;

function multiplicarConGlobales() {
    console.log(aGlobal * bGlobal);
}
multiplicarConGlobales();

// Función que devuelve un valor y luego lo usa fuera de la función
function multiplicarYDevolver() {
    let a = 5;
    let b = 8;
    return a * b;
}

let resultado = multiplicarYDevolver();
console.log(resultado + 5); // Imprimir el resultado sumado a 5

// Función con una validación que devuelve un valor booleano
function validarDato() {
    if (aGlobal === 5) {
        return true;
    } else {
        return false;
    }
}

let dato = validarDato();
console.log(dato); // true

File: funciones.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    // Multiplicación de dos vectores
    vector<int> a = {1, 2, 3, 4, 5};
    vector<int> b = {6, 7, 8, 9, 10};
    vector<int> c;

    // Multiplicando los elementos de los dos vectores
    for (int i = 0; i < a.size(); i++) {
        c.push_back(a[i] * b[i]);
    }

    // Imprimir el resultado de la multiplicación
    for (int i = 0; i < c.size(); i++) {
        cout << c[i] << " ";
    }
    cout << endl;

    // Funciones

    // Definición y llamada
    void mostrarTexto() {
        cout << "Hola" << endl;
    }

    mostrarTexto();

    // Ámbito de las variables
    int x = 5, y = 8;

    void multiplicar() {
        // Usamos las variables globales 'x' y 'y'
        cout << x * y << endl;
    }

    multiplicar();

    // Usar return
    int multiplicarReturn() {
        int a = 5, b = 8;
        return a * b;
    }

    int resultado = multiplicarReturn();
    cout << resultado + 5 << endl;

    // Trabajar con valores devueltos
    bool validarDato() {
        int a = 5;
        if (a == 5) {
            return true;
        } else {
            return false;
        }
    }

    bool dato = validarDato();
    cout << (dato ? "true" : "false") << endl;

    return 0;
}